feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIcalls.java

```diff
@@ -0,0 +1,158 @@
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Optional;
+
+import static com.sun.org.apache.xml.internal.utils.LocaleUtility.EMPTY_STRING;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class CraterAPIcalls {
+
+    private final CraterProperties properties = CraterProperties.getInstance();
+    CraterLogin craterLogin = new CraterLogin();
+    private final String auth = craterLogin.getToken();
+
+    private static final Logger logger = LoggerFactory.getLogger(CraterAPIcalls.class);
+
+    public CraterAPIcalls() {
+    }
+
+    public HttpURLConnection getConnection(String api, String httpMethod, String auth) throws IOException {
+        URL url = new URL(properties.getUrl() + "/api/v1/" + api);
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        con.setRequestMethod(httpMethod);
+        con.setRequestProperty("Authorization", "Bearer " + auth);
+        con.setRequestProperty("Content-Type", properties.getContent_type());
+        con.setRequestProperty("Accept", properties.getAccept());
+        con.setRequestProperty("company", properties.getCompany());
+        con.setDoOutput(Boolean.parseBoolean(properties.getSetDoOutput()));
+        con.setDoInput(Boolean.parseBoolean(properties.getSetDoInput()));
+        return con;
+    }
+
+    public String login_verification(String auth) throws Exception {
+        CloseableHttpClient httpClient = HttpClients.createDefault();
+        HttpGet request = new HttpGet(properties.getUrl() + "/api/v1/auth/check");
+
+        request.addHeader("Content-Type", properties.getContent_type());
+        request.addHeader("Accept", properties.getAccept());
+        request.addHeader("company", properties.getCompany());
+        request.addHeader("Authorization", "Bearer " + auth);
+
+        try (CloseableHttpResponse response = httpClient.execute(request)) {
+            HttpEntity entity = response.getEntity();
+            String result = EntityUtils.toString(entity);
+            return ((result.charAt(0) == '1') ? "Authenticated" : "Not Authenticated");
+        }
+    }
+
+    public void create_customer(OpenMRSPatientFullRepresentation patientFR, String auth) throws Exception {
+            String name = getname(patientFR);
+            String uuid = getuuid(patientFR);
+
+            HttpURLConnection con = getConnection("customers", "POST", auth);
+
+            JSONObject parameters = new JSONObject();
+            parameters.put("name", name);
+            parameters.put("contact_name", uuid);
+            parameters.put("currency_id", properties.getCurrencyid());
+            String jsonInputString = parameters.toString();
+
+            try (OutputStream os = con.getOutputStream()) {
+                byte[] inputString = jsonInputString.getBytes(UTF_8);
+                os.write(inputString, 0, inputString.length);
+            }
+            con.getInputStream();
```

Comment:
Don't these streams need to be closed after the create_customer( ) operation is done? Or do these remain open throughout the lifecycle of this class?

Reply:
Streams have been closed. They don't need to remain open.